cmd/nri-kubernetes/main.go

```diff
@@ -244,7 +258,20 @@ func setupKubelet(c *config.Config, clients *clusterClients) (*kubelet.Scraper,
 		Kubelet:  clients.kubelet,
 		CAdvisor: clients.cAdvisor,
 	}
-	ksmScraper, err := kubelet.NewScraper(c, providers, kubelet.WithLogger(logger))
+
+	scraperOpts := []kubelet.ScraperOpt{kubelet.WithLogger(logger)}
+
+	if c.NamespaceSelector != nil {
+		nsFilter := discovery.NewNamespaceFilter(
+			c.NamespaceSelector,
+			clients.k8s,
+			storer.NewInMemoryStore(c.Interval, c.Interval+storer.NamespaceStoreExtraInterval, logger),
+			logger,
+		)
+		scraperOpts = append(scraperOpts, kubelet.WithFilterer(nsFilter))
+	}
```

Comment:
would it make sense a helper function to avoid repeating the `nsFilter` creation?

```go
nsFilter := buildNamespaceFilter(c, clients.k8s, logger)
if nsFilter != nil {
    scraperOpts = append(scraperOpts, kubelet.WithFilterer(nsFilter))
}
```

Reply:
sure.